src/providers/abstract-provider.ts

```diff
@@ -425,117 +425,117 @@ export type PerformActionRequest =
     | {
           method: 'broadcastTransaction';
           signedTransaction: string;
-          shard: string;
+          zone: Zone;
       }
     | {
           method: 'call';
           transaction: PerformActionTransaction;
           blockTag: BlockTag;
-          shard?: string;
+          zone?: Zone;
       }
     | {
           method: 'chainId';
-          shard?: string;
+          zone?: Zone;
       }
     | {
           method: 'estimateGas';
           transaction: PerformActionTransaction;
-          shard?: string;
+          zone?: Zone;
       }
     | {
           method: 'getBalance';
           address: string;
           blockTag: BlockTag;
-          shard: string;
+          zone: Zone;
       }
     | {
           method: 'getOutpointsByAddress';
           address: string;
-          shard: string;
+          zone: Zone;
       }
     | {
           method: 'getBlock';
           blockTag: BlockTag;
           includeTransactions: boolean;
-          shard: string;
+          shard: Shard;
       }
     | {
           method: 'getBlock';
           blockHash: string;
           includeTransactions: boolean;
-          shard: string;
+          shard: Shard;
       }
     | {
           method: 'getBlockNumber';
-          shard?: string;
+          shard?: Shard;
       }
     | {
           method: 'getCode';
           address: string;
           blockTag: BlockTag;
-          shard: string;
+          shard: Shard;
       }
     | {
           method: 'getGasPrice';
           txType: boolean;
-          shard?: string;
+          zone?: Zone;
       }
     | {
           method: 'getLogs';
           filter: PerformActionFilter;
-          shard: string;
+          shard: Shard;
       }
     | {
           method: 'getMaxPriorityFeePerGas';
-          shard?: string;
+          zone?: Zone;
       }
     | {
           method: 'getStorage';
           address: string;
           position: bigint;
           blockTag: BlockTag;
-          shard: string;
+          shard: Shard;
```

Comment:
This method relates to returning smart contract data at some storage position so this should be zone